R/get_df.R

```diff
@@ -5,11 +5,21 @@
 #'   from regression models.
 #'
 #' @param x A statistical model.
-#' @param type Can be `"residual"`, `"model"` or `"analytical"`. `"residual"`
-#' tries to extract residual degrees of freedoms. If residual degrees of freedom
-#' could not be extracted, returns analytical degrees of freedom, i.e. `n-k`
-#' (number of observations minus number of parameters). `"model"` returns
-#' model-based degrees of freedom, i.e. the number of (estimated) parameters.
+#' @param type Can be `"residual"`, `"wald"`, `"normal"`, `"analytical"`, or
+#'   `"model"`.
+#'
+#' - `"residual"` tries to extract residual degrees of freedoms. If residual
+#'   degrees of freedom could not be extracted, returns analytical degrees of
+#'   freedom, i.e. `n-k` (number of observations minus number of parameters).
+#' - `"wald"` for models with z-statistic, returns `"Inf"`. Else, tries to
+#'   extract residual degrees of freedoms. If residual degrees of freedom could
+#'   not be extracted, returns `"Inf"`.
+#' - `"analytical"` returns analytical degrees of freedom, i.e. `n-k`
```

Comment:
What's the difference between residual and analytical? They sound the same to me from this description

Reply:
analytical is always `n_obs() - n_param()` (n-k). not sure if which cases this may differ from residual df?